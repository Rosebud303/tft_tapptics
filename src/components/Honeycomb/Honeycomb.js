import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setCurrentRosterIndex } from '../../actions'

export class Honeycomb extends Component {



  render() {
    {
      if(this.props.img !== '') {
        return (
          <div className='tiles-main' style={{backgroundColor: 'rgba(63, 81, 181, 0.58)'}} onDragOver={() => this.props.setRosterIndex(this.props.idx)}>
            <img className='tile-img' src={this.props.img}/>
          </div>
        )
      } else {
        return (
          <div className='tiles-main' onDragOver={() => this.props.setRosterIndex(this.props.idx)}>
          </div>
        )
      }
    }
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setRosterIndex: (idx) => dispatch(setCurrentRosterIndex(idx))
})

export default connect(null, mapDispatchToProps)(Honeycomb)
