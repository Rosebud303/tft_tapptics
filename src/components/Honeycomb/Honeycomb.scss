.tiles-main {
  margin: 15px;
  display: inline-block;
  height: 125px;
  width: 14%;
  clip-path: polygon(50% 0%, 100% 25%, 100% 75%, 50% 100%, 0% 75%, 0% 25%);
  position: relative;
  background: #9e9e9e40;
}

.tile-img {
  height: 80px;
  width: 80%;
  position: absolute;
}