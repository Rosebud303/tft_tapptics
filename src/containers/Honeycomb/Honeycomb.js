import React, { Component } from 'react'
import { HexGrid, Layout, Hexagon, Text, Pattern, Path, Hex } from 'react-hexgrid';

export default class Honeycomb extends Component {
  render() {
    return (
      <div className="Honeycomb">
        <HexGrid width={1000} height={600} viewBox="-50 -50 100 100">
          <Layout size={{ x: 12, y: 12 }} flat={false} spacing={1.1} origin={{ x: -10, y: -10 }}>
            <Hexagon q={-1} r={-1} s={0} fill="pat-1"  />
            <Hexagon q={0} r={-1} s={0} fill="pat-2" />
            <Hexagon q={1} r={-1} s={0} fill="pat-3"/>
            <Hexagon q={2} r={-1} s={0} fill="pat-4" />
            <Hexagon q={-1} r={0} s={0} fill="pat-5" />
            <Hexagon q={1} r={0} s={0} fill="pat-6" />
            <Hexagon q={-2} r={0} s={0} fill="pat-7" />
            <Hexagon q={0} r={0} s={0} fill="pat-8" />
            <Hexagon q={-2} r={1} s={0} fill="pat-9" />
            <Hexagon q={-1} r={1} s={0} fill="pat-10" />
            <Hexagon q={0} r={1} s={0} fill="pat-11" />
            <Hexagon q={1} r={1} s={0} fill="pat-12" />
            <Path start={new Hex(0, 0, 0)} end={new Hex(-2, 0, 1)} />
          </Layout>
          <Pattern className="pattern" id="pat-1" link="https://i.imgur.com/MTvNXB1.png" />
          <Pattern id="pat-2" link="https://i.imgur.com/EkVoOjj.png" className="" />
        </HexGrid>
      </div>
    )
  }
}
