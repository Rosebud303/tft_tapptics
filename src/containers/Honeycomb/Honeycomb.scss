.Honeycomb {

  svg polygon {
    fill: hsla(0, 0%, 100%, 0.5);
  }
}