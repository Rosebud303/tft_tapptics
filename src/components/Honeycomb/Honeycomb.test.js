import React from 'react';
import { shallow } from 'enzyme';
import Honeycomb from './Honeycomb';

describe('Honeycomb', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Honeycomb />);
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});