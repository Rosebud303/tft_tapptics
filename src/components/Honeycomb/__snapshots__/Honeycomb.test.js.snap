// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Honeycomb should match the snapshot with all the data passed in 1`] = `
<Layout
  className="Honeycomb"
  flat={false}
  origin={
    Object {
      "x": -10,
      "y": -10,
    }
  }
  size={
    Object {
      "x": 13,
      "y": 13,
    }
  }
  spacing={1.1}
>
  <Hexagon
    fill="pat-1"
    q={-1}
    r={-1}
    s={0}
  />
  <Hexagon
    fill="pat-2"
    q={0}
    r={-1}
    s={0}
  />
  <Hexagon
    fill="pat-3"
    q={1}
    r={-1}
    s={0}
  />
  <Hexagon
    fill="pat-4"
    q={2}
    r={-1}
    s={0}
  />
  <Hexagon
    fill="pat-5"
    q={-1}
    r={0}
    s={0}
  />
  <Hexagon
    fill="pat-6"
    q={1}
    r={0}
    s={0}
  />
  <Hexagon
    fill="pat-7"
    q={-2}
    r={0}
    s={0}
  />
  <Hexagon
    fill="pat-8"
    q={0}
    r={0}
    s={0}
  />
  <Hexagon
    fill="pat-9"
    q={-2}
    r={1}
    s={0}
  />
  <Hexagon
    fill="pat-10"
    q={-1}
    r={1}
    s={0}
  />
  <Hexagon
    fill="pat-11"
    q={0}
    r={1}
    s={0}
  />
  <Hexagon
    fill="pat-12"
    q={1}
    r={1}
    s={0}
  />
</Layout>
`;
